fred/libraries/createembed.py

```diff
@@ -388,9 +388,11 @@ async def mod_embed(
     if not mods:
         return None, None, None
 
-    single_mod = common.mod_name_eq((mod := mods[0])["name"], name) or len(mods) == 1
+    single_mod = len(mods) == 1 
+        or (common.mod_name_eq((mod := mods[0])["name"], name) and not common.mod_name_eq((mod := mods[1])["name"], name))
```

Comment:
I am pretty sure this won't work because putting `len(mods) == 1` first can short circuit and leave `mod` unassigned. Using the assignment operator twice makes the logic harder to follow and might also break something (wish we had tests). Ideally this could be refactored to not use the assignment operator.

Reply:
Yep, makes sense. My python is rusty enough that I haven't done any since... um... well, before the assignment operator was introduced in 3.8! Will have a look at reworking that soon.